atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -46,6 +46,25 @@ def __init__(self, tasker, workflow, hub, target=None, koji_ssl_certs_dir=None):
             }
         self.xmlrpc = create_koji_session(hub, koji_auth_info)
 
+    def get_patched_release(self, original_release, increment=False):
+        if original_release.count('.') > 1:
+            release, suffix, remainder = original_release.split('.', 2)
+        else:
+            remainder = ''
+            if original_release.count('.') == 1:
+                release, suffix = original_release.split('.', 1)
+            else:
+                release = original_release
+                suffix = ''
+
+        result = [str(int(release) + 1)] if increment else [release]
```

Comment:
I find this quite hard to follow because of all the cases for splitting. How about this?
```
parts = original_release.split('.', 2) + [None, None]
release, suffix, rest = parts[:3]
if increment:
    # Increment first part as a number
    release = str(int(release) + 1)

# Remove second part if it's a number
if suffix is not None and suffix.isdigit():
    suffix = None

# Recombine the parts
return '.'.join([part for part in [release, suffix, rest]
                 if part is not None])
```